src/App.tsx

```diff
@@ -1,56 +1,80 @@
-import React from 'react';
+import React, {
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import './App.scss';
+import { debounce } from 'lodash';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+import { Person } from './types/Person';
+
+const LIST_UPDATE_DELAY = 1000;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isSelected, setIsSelected] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, LIST_UPDATE_DELAY), [],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, peopleFromServer]);
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
+    applyQuery(event.target.value);
+    setQuery(event.target.value);
+  }
+
+  const handleSelectedPerson = useCallback((person: Person) => {
+    setAppliedQuery(person.name);
+    setQuery(person.name);
+    setSelectedPerson(person);
+    setIsSelected(true);
+  }, []);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {
+          isSelected
```

Comment:
```suggestion
          selectedPerson
```
and remove `isSelected` state at all